transhumandesign/kag-base
# [added] Water dripping effect for filled sponge and for runners who got out of water

## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description
```
[added] Water dripping effect
```

This PR adds a sprite script `WaterDrops.as` which is added to all runner classes.
It creates water particles and water drip sound if the blob has tag `"wet"`.

`RunnerDefault.as` will tag the runner `"wet"` if in water or if he has been in water in the past 5 seconds, otherwise untags it.

The effect only plays out if "faster graphics" is turned off.

The sounds have been extracted from here: https://pixabay.com/de/sound-effects/water-dripping-on-wateraif-14581/

Tested in offline and online, looks to be working well. The only thing I'm worrying is if the effect works correctly with many players in a server. Maybe checks for "is client", "is server", "is my player", etc. have to be added.

## Alternative solutions

As an alternative, instead of adding a sprite script to each .cfg file, it is probably possible to add it from within `RunnerDefault.as`.

## Steps to Test or Reproduce

Have "faster graphics" turned off.
Enter water and leave the water, notice water dripping off your character.

Throw a sponge in water and wait for it to be full.
Get the sponge out of water.
Notice the sponge will have water dripping off it.
